kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -4,28 +4,59 @@
  */
 package io.strimzi.kafka.agent;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.yammer.metrics.core.Gauge;
 import com.yammer.metrics.core.Metric;
 import com.yammer.metrics.core.MetricName;
 import com.yammer.metrics.core.MetricsRegistry;
 import com.yammer.metrics.core.MetricsRegistryListener;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A very simple Java agent which polls the value of the {@code kafka.server:type=KafkaServer,name=BrokerState}
  * Yammer Metric and once it reaches the value 3 (meaning "running as broker", see {@code kafka.server.BrokerState}),
  * creates a given file.
  * The presence of this file is tested via a Kube "exec" readiness probe to determine when the broker is ready.
+ * It also exposes a REST endpoint for broker metrics.
+ * <dl>
+ *     <dt>{@code GET /v1/broker-state}</dt>
+ *     <dd>Reflects the BrokerState metric, returning a JSON response e.g. {"brokerState": 3}.
+ *      If broker state is RECOVERY(2), it includes remainingLogsToRecover and remainingLogsToRecover in the response e.g.
+ *      {"brokerState": 2
+ *       "recovery": {
+ *          "remainingLogsToRecover": 123
+ *          "remainingLogsToRecover": 456
+ *        }
+ *      }</dd>
+ * </dl>
  */
 public class KafkaAgent {
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAgent.class);
+    private static final String BROKER_STATE_PATH = "/v1/broker-state";
+    private static final int HTTPS_PORT = 8443;
+    private static final long GRACEFUL_SHUTDOWN_TIMEOUT_MS = 60 * 1000;
```

Comment:
The default graceful shutdown period on Kubernetes is 30 seconds. That is what most users will be using. So I think it makes little sense to have this set to anything higher than that.